### PR TITLE
TSK-1390: Add SPI to allow task processing before creation

### PR DESCRIPTION
**Pull request by gitgoodjhe**
Sonarcloud: https://sonarcloud.io/dashboard?branch=TSK_1390&id=gitgoodjhe_taskana

### CODE REVIEW

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java [hunk 1]
```diff
@@ -209,6 +212,10 @@ public Task createTask(Task taskToCreate)
       workbasketService.checkAuthorization(
           task.getWorkbasketSummary().getId(), WorkbasketPermission.APPEND);
 
+      if (CreateTaskPreprocessorManager.isCreateTaskPreprocessorEnabled()) {
```
Comment (holgerhagen) [inline lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java:215]: Why did you put it here? I would have expected it even before the opening of the connection to the database...

File: lib/taskana-core/src/main/java/pro/taskana/spi/taskpreprocessing/api/CreateTaskPreprocessor.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package pro.taskana.spi.taskpreprocessing.api;
```
Comment (holgerhagen) [inline lib/taskana-core/src/main/java/pro/taskana/spi/taskpreprocessing/api/CreateTaskPreprocessor.java:1]: I would prefer the more simple package pro.taskana.spi.task.
And do we really need ".api"? Isn't SPI already enough?

**Review by holgerhagen — CHANGES_REQUESTED** [review]

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java (hunk 1 above)
Comment (gitgoodjhe) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java:220]: I wasn't sure if the workbasket/summary would definately never be needed in the SPI method....changed it to before opening the connection of the database now

File: lib/taskana-core/src/main/java/pro/taskana/spi/taskpreprocessing/api/CreateTaskPreprocessor.java (hunk 2 above)
Comment (gitgoodjhe) [reply lib/taskana-core/src/main/java/pro/taskana/spi/taskpreprocessing/api/CreateTaskPreprocessor.java]: changed the package to pro.taskana.spi.task. api/internal packages to be consistent with history and task trouting spis. Do you want all of them restructured to without api/internal packages?

**Review by holgerhagen — APPROVED** [review]

**Pull request merged**